File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/spotcleaning/SpotCleaningServiceFactory.java
package com.neatorobotics.sdk.android.robotservices.spotcleaning;

/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public class SpotCleaningServiceFactory {
    public static SpotCleaningService get(String serviceVersion) {
        if("basic-1".equalsIgnoreCase(serviceVersion)) return new SpotCleaningBasic1Service();
        else if("basic-2".equalsIgnoreCase(serviceVersion)) return new SpotCleaningBasic2Service();
        else if("minimal-2".equalsIgnoreCase(serviceVersion)) return new SpotCleaningMinimal2Service();
        else if("micro-2".equalsIgnoreCase(serviceVersion)) return new SpotCleaningMicro2Service();
        else return null;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/spotcleaning/SpotCleaningServiceFactory.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.spotcleaning

import com.neatorobotics.sdk.android.robotservices.RobotServices

object SpotCleaningServiceFactory {
    fun get(serviceVersion: String): SpotCleaningService? {
        return when {
            RobotServices.VERSION_BASIC_1.equals(serviceVersion, ignoreCase = true) -> SpotCleaningBasic1Service()
            RobotServices.VERSION_BASIC_3.equals(serviceVersion, ignoreCase = true) -> SpotCleaningBasic3Service()
            RobotServices.VERSION_MINIMAL_2.equals(serviceVersion, ignoreCase = true) -> SpotCleaningMinimal2Service()
            RobotServices.VERSION_MICRO_2.equals(serviceVersion, ignoreCase = true) -> SpotCleaningMicro2Service()
            else -> null
        }
    }
}
